test/acceptance/keyword_argument_matching_test.rb

```diff
@@ -15,7 +17,9 @@ def test_should_match_hash_parameter_with_keyword_args
     test_result = run_as_test do
       mock = mock()
       mock.expects(:method).with(:key => 42)
-      mock.method({ :key => 42 }) # rubocop:disable Style/BracesAroundHashParameters
+      DeprecationDisabler.disable_deprecations do
```

Comment:
> For my understanding, this is purely to prettify the test output right? 

Yes.

> Is it worth turning off Mocha's deprecation printing by default in the entire test suite instead? Since we have the messages, we can still test deprecation text for each test. And we can turn on printing for a unit test where we are checking that deprecations are stdout.


I think the reason I haven't done that in the past is that I'd like to see any _unexpected_ deprecation warnings when running the tests. I'll give it some more thought.

> I'm just thinking this will eventually become dead code if the [planned Mocha v2](https://github.com/freerange/mocha/pull/562#issuecomment-1277569706) works out.

Yeah, that's a fair point, although there will be some work to do when the deprecation warnings are removed and I'm not sure this will add much to that.

Reply:
> Yeah, that's a fair point, although there will be some work to do when the deprecation warnings are removed and I'm not sure this will add much to that.

It's not a major concern, but one would have to manually remember which DeprecationDisablers to remove, as they wouldn't fail or raise anything. I think it'd be easy to forget one. Whereas, if they weren't there in the first place... 😄

Reply:
Either way, not a blocker for this PR, I'd think it better to do that separately anyway 😄

Reply:
I've extracted this into a separate issue: https://github.com/freerange/mocha/issues/565

Reply:
I've decided to address the concern in this PR by adding assertions for the specific deprecation warnings. This should mean it's less likely we'll leave calls to `DeprecationDisabler.disable_deprecations` littered around the codebase once the relevant deprecation warnings are removed. It's unfortunate that this makes the tests a bit less readable and verbose, but until we come up with a better generic approach in #565, I think this is a reasonable solution. See https://github.com/freerange/mocha/pull/563/commits/40cec7ab7f92c23c205d85ba85223c54f23b5fdb.